bonfire/openshift.py

```diff
@@ -732,3 +732,33 @@ def _pod_found():
     waiter.wait_for_ready(remaining_time)
 
     return pod_name
+
+
+def wait_on_reservation(res_name, timeout):
+    log.info("waiting for reservation '%s' to get picked up by operator", res_name)
+
+    def _find_reservation():
+        res = get_json("reservation", name=res_name)
+        try:
+            return res["status"]["namespace"]
+        except (KeyError, IndexError):
+            return False
+
+    ns_name, elapsed = wait_for(
+        _find_reservation,
+        num_sec=timeout,
+        message=f"wait for namespace to be owned by reservation '{res_name}'",
+    )
+    return ns_name
+
+
+def get_reservation_by_requester(requester):
```

Comment:
I think we should allow for ns lookup/ns extension using:

* reservation name
* namespace name (find the reservation associated with namespace "X")

Reply:
Changed reservation lookup to either use reservation name or attached namespace. Left out retrieving reservations via requester, and instead added some additional filtering capabilities to "list" to print out reservations via "--mine" (uses oc whoami) or "--requester".